src/main/java/jenkins/branch/BranchNameContributor.java

```diff
@@ -64,8 +64,6 @@ public void buildEnvironmentFor(Job j, EnvVars envs, TaskListener listener) thro
                 envs.put("BRANCH_NAME", head.getName());
                 if (branch.getAction(PrimaryInstanceMetadataAction.class) != null) {
                     envs.put("BRANCH_IS_PRIMARY", "true");
-                } else {
-                    envs.put("BRANCH_IS_PRIMARY", "false");
```

Comment:
Alternatively, which was my initial suggestion:
```suggestion
                } else {
                    envs.put("BRANCH_IS_PRIMARY", "");
```

This would at least allow scripts to distinguish between the `false` case and one where an older version of the plugin is in use.

Reply:
I don't inherently have an issue with this proposal. The truthy part of a "false" value can be confusing (as can be treating any value as true).

Reply:
If you prefer my suggestion in the above comment to the implementation currently proposed in the PR, I can update it.

Thanks for the comment!

(Sorry for the delay, it wasn't clear to me if others would want to chime in first.)

Reply:
What happens if one attempts to us this variable inside the pipeline in a string interpolation for example:
```
sh "BRANCH_IS_PRIMARY = ${env.BRANCH_IS_PRIMARY}"
sh "BRANCH_IS_PRIMARY = $BRANCH_IS_PRIMARY"
```
Won't the result in an error for the false case if we take this change?

Reply:
You would get an error unless you change it to what I think you mean:

```groovy
sh "BRANCH_IS_PRIMARY=${env.BRANCH_IS_PRIMARY}"
sh "BRANCH_IS_PRIMARY=$BRANCH_IS_PRIMARY"
```

With this, neither the empty string nor the `"false"` case will result in an error, unless I'm missing something?

Reply:
Hello! I was just checking whether the PR is OK to merge as-is, or if you'd rather set the empty string as the alternative suggested in this comment, and I just realized I may have misunderstood @bitwiseman's question. Indeed, if someone tries to use a variable that isn't set, there will be an error. I wonder if it's an issue since there is the same possibility with [other `...; else unset` variables](https://github.com/jenkinsci/branch-api-plugin/blob/master/src/main/resources/jenkins/branch/BranchNameContributor/buildEnv.properties).

This would mean that the variable should only be checked via `env.BRANCH_IS_PRIMARY` and not `BRANCH_IS_PRIMARY` (unless not using string interpolation). An alternative to not setting it in the false case would be to set it to the empty string, as proposed here.

Reply:
> Indeed, if someone tries to use a variable that isn't set, there will be an error.